### ISSUE TITLE
Project reference issue (No project references found)

### ISSUE THREAD

**Issue by masifi** [stryker-mutator/stryker-net#461]
I have used this very basic config. in a single project
```
 <!-- example.csproj -->
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
    <PackageReference Include="xunit" Version="2.4.1" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1" />
    <PackageReference Include="StrykerMutator.Core" Version="0.9.0" />
    <PackageReference Include="StrykerMutator.DotNetCoreCli" Version="0.9.0" />
  </ItemGroup>
</Project>
```

Later I ran the same commands as in docs.
`dotnet stryker`
`dotnet stryker --project-file example.csproj`
`dotnet stryker --config-file-path stryker.config.json`
`dotnet stryker --test-runner "vstest" --reporters "['Html', 'ConsoleReport', 'ConsoleProgressBar']" --log-console "debug"`
Always, I see  the same logs and mutation won't start, however the unit test has no problem and it is running fine.
```
# terminal output 1
Stryker.NET failed to mutate your project. For more information see the logs below:

Project reference issue.

No project reference matched your --project-file=evaluator.csproj
Choose one of the following references:
```
```
# terminal output 2
Stryker.NET failed to mutate your project. For more information see the logs below:

Project reference issue.

No project references found. Please add a project reference to your test project and retry.
```
It would be appreciated if you have any suggestion that would help.

**Comment by richardwerkman** [comment]
Hi Masifi, Stryker is trying to find out what project should be mutated. It does by scanning for project references in your test csproj file. As I see in your example.csproj you have no project references in your project file. Thus stryker cannot find a project to mutate.

Stryker needs to find an external project to mutate as it should not mutate your test code, but your logic. It is best practice to place your logic outside of your test project.

You can see an example project file in our [integration test](https://github.com/stryker-mutator/stryker-net/blob/master/integrationtest/TargetProjects/NetCoreTestProject.XUnit/NetCoreTestProject.XUnit.csproj).

When you add the project reference to your test project stryker should run fine :+1:

**Comment by masifi** [comment]
@richardwerkman thanks a lot, it really worked, it was my fault such that I've put the logic inside test project, as you suggested I've separated the project and referenced the "business logic" project from test project then it worked perfectly  👍

**Comment by richardwerkman** [comment]
Good to hear! Have fun improving your tests :ok_hand:

**Comment by nohwnd** [comment]
Had the same issue, the project needed to be re-built after I added the reference.

**Comment by Mobrockers** [comment]
Hey @nohwnd 

We no longer provide stryker as a dotnet core cli extensibility model tool, it is packaged as a dotnet global tool since version 0.10.0. So no references need to be added to your projects.

Do you have any issues using the global tool?

**Comment by nohwnd** [comment]
I do not, I was referring to the: 

> Stryker is trying to find out what project should be mutated. It does by scanning for project references in your test csproj file. As I see in your example.csproj you have no project references in your project file.

I was using just a single test project with no Project dependencies. After reading this I added a ClassLibrary project to hold some code. I added reference to that ClassLibrary from my test project and executed `dotnet styker` which gave me the same error as before.

After this I hit build, and then `dotnet styker` was able to run successfully. So I was just commenting that a rebuild was needed, even though I had Project references.
